### ISSUE TITLE
Inline SVG won't render in JSX

### ISSUE THREAD

**Issue by jpopesculian** [cyclejs/core#254]
I wanted to use a logo as an inline svg, so I created a component (pasted below), to inject it. However although the HTML gets outputted correctly (I can go in the console > select Edit as HTML > add a newline after the SVG and press ctrl+enter and it renders) the generated SVG doesn't render. Is there something I don't know about dynamic SVG rendering or is it possibly the hJSX or is this something that cycle js can't do?

```javascript
import {hJSX} from '@cycle/dom'
import isolate from '@cycle/isolate'
import {Observable} from 'rx'

const intent = (DOM) => {
  return {}
}

const model = (actions) => {
  return Observable.just({})
}

const view = (state$) => {
  return state$.map((state) => {
    return (
        <div className="logo">
            <svg attributes={
                {
                    "xmlns:svg": "http://www.w3.org/2000/svg",
                    "xmlns": "http://www.w3.org/2000/svg",
                    "width": "104.5",
                    "height": "35.7",
                    "version": "1.1"
                }
            }>
                <g attributes={{transform: "translate(-119.87724,-519.68907)"}}>
                    <path attributes={{d: "m119.9 555.2 0-35 5 0 0 35-5 0M141.7 535.2l1.3 0c1.1 0 2.1-0.1 2.8-0.2 0.7-0.1 1.4-0.4 1.9-0.7 0.6-0.4 1-0.9 1.2-1.5 0.2-0.7 0.4-1.5 0.4-2.5 0-1.2-0.1-2.3-0.3-3-0.2-0.8-0.5-1.4-1-1.9-0.5-0.5-1.1-0.8-1.9-1-0.8-0.2-1.8-0.3-3.1-0.3l-1.2 0 0 11.3m-5 20 0-35 8.8 0c5.8 0 8.7 3.3 8.8 10 0 2.9-0.8 5-2.3 6.5-1.5 1.5-3.7 2.2-6.5 2.2l-3.7 0 0 16.3-5 0"}}/>
                    <g attributes={{transform: "translate(3.75,-0.17857143)"}}>
                        <path attributes={{d: "m162.2 555.5 0-35 3.8 0 8.8 21.6 5 13.4-3.6 0-8.9-23.3 0 23.3-5 0M172.3 520.4l3.8 0 8.8 21.6 0-21.6 5 0 0 35-3.6 0-8.9-23.3"}}/>
                    </g>
                    <path attributes={{d: "m215.6 555.4c-3.2 0-5.6-0.9-7.2-2.8-1.7-1.8-2.6-4.4-2.8-7.6l5-1.2c0.2 5 1.9 7.5 5 7.5 2.5 0 3.7-1.2 3.8-3.7 0-0.7-0.1-1.4-0.3-2.1-0.2-0.6-0.5-1.2-0.9-1.7-0.4-0.5-0.8-1-1.1-1.3-0.3-0.3-0.8-0.7-1.4-1.2l-6.2-5c-1.1-0.9-2.1-2-2.8-3.4-0.7-1.4-1-3-1-4.7 0-2.6 0.8-4.6 2.4-6.2 1.6-1.5 3.7-2.3 6.4-2.3 2.8 0 5.1 0.7 6.8 2.1 1.7 1.4 2.7 3.3 3.1 5.9 0 0.3 0.1 0.7 0.1 1.1l-5 1.3c0-0.2 0-0.6-0.1-1-0.2-1.6-0.7-2.9-1.5-3.8-0.8-1-1.9-1.4-3.4-1.4-1.2 0-2.1 0.3-2.8 1-0.7 0.7-1 1.6-1 2.8 0 2 0.8 3.6 2.5 4.9l6.3 5c3.3 2.7 5 5.9 5 9.7 0 2.6-0.8 4.6-2.4 6-1.6 1.4-3.7 2.2-6.4 2.2"}}/>
                </g>
            </svg>
        </div>
    )
  })
}

const Logo = ({DOM, prop$}) => {
  const state$ = model(intent(DOM))
  return {
    value$: state$,
    DOM: view(state$)
  }
}

export default isolate(Logo)
```

**Comment by laszlokorte** [comment]
I think it's the `hJSX` that does not work for svg.

The hyperscript `h()` function does not work for svg either because the namespaces are not set correctly.
Instead you need to use the `svg()` function that virtual-dom provides. I assume `hJSX` does not check for svg tag names and does not delegate to `svg()` but just to `h()`

**Comment by staltz** [comment]
> I assume hJSX does not check for svg tag names and does not delegate to svg() but just to h()

And that would be the fix :)

**Comment by psychowico** [comment]
is this possible to fix at all? hJSX function get info about specific tag.
how we could now that this is svg tag? some tags have same names in svg and pure html, e.g. `<a>`

https://developer.mozilla.org/en-US/docs/Web/SVG/Element

**Comment by staltz** [comment]
Fixed with Cycle DOM v9.4.0
